apps/els_lsp/src/els_sheldon_diagnostics.erl

```diff
@@ -0,0 +1,119 @@
+%%==============================================================================
+%% Compiler diagnostics
+%%==============================================================================
+
+-module(els_sheldon_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+
+-export([ init/0
+        , is_default/0
+        , run/1
+        , source/0
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+
+-include("els_lsp.hrl").
+-include_lib("kernel/include/logger.hrl").
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+
+-spec init() -> ok.
+init() ->
+  application:ensure_all_started(sheldon),
+  ok.
+
+-spec is_default() -> boolean().
+is_default() ->
+  false.
+
+-spec run(uri()) -> [els_diagnostics:diagnostic()].
+run(Uri) ->
+  case els_utils:project_relative(Uri) of
+    {error, not_relative} -> [];
+    RelFile ->
+      try
+        RegEx = "[_@./#&+-=*]",
```

Comment:
Not being familiar with Sheldon, can you explain this regex?

Reply:
This regex provide logic for rebar plugin for skip lines in source code, eg for `elvis.config` etc.